pandas/io/common.py

```diff
@@ -726,6 +726,12 @@ def __init__(
         self.archive_name = archive_name
         self.multiple_write_buffer: Optional[Union[StringIO, BytesIO]] = None
 
+        # add csv file name like gz or bz2
+        if archive_name is None and isinstance(file, (os.PathLike, str)):
+            archive_name = os.path.basename(file)
```

Comment:
probably better to replace `os.path.basename` with `stringify_path` (to handle PathLike objects). That might also fix the failing `test_to_csv_compression_encoding_gcs` test.

Reply:
its strange that it passed this test on my win10 laptop.
![image](https://user-images.githubusercontent.com/15418465/107300582-c6fc1980-6ab4-11eb-9cbe-f427ca2f022f.png)

Reply:
sorry, you probably still need `basename` to avoid having the entire path within the archive

Reply:
start get different ci errors, does my commit affect it?

Reply:
Can you add a small test case. For example, write a compressed CSV file and then open it with zipfile.ZipFile to check the name.

Reply:
add test case and finish pre commit, a new PR is opened #40387